src/NetAnalyzers/CSharp/Microsoft.CodeQuality.Analyzers/QualityGuidelines/CSharpMarkMembersAsStatic.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeQuality.Analyzers.QualityGuidelines;
+
+namespace Microsoft.CodeQuality.CSharp.Analyzers.QualityGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpMarkMembersAsStaticAnalyzer : MarkMembersAsStaticAnalyzer
+    {
+        protected override bool SupportStaticLocalFunctions(ParseOptions parseOptions)
+            => ((CSharpParseOptions)parseOptions).LanguageVersion >= LanguageVersion.CSharp8;
```

Comment:
Out of scope for the PR, but I'd love to have `ILanguageFeatureSupportService` with language specific implementations, and do something like `compilation.GetService<ILanguageFeatureSupportService>().IsFeatureSupported(LanguageFeature.StaticLocalFunctions);`

Benefits are:

- This will eliminate the need for language-specific classes for some analyzers.
- One central place to decide whether a feature is supported.

But I'm not sure if this is easily approachable, or if it's worth doing.